src/views/data.js

```diff
@@ -0,0 +1,99 @@
+const util = require('../lib/util');
+const messagesUtil = require('../lib/messages');
+
+const defaultSummary = { total: '--', time: '--' }
+
+function processType(type, indexedMessages) {
+    const messages = indexedMessages[type];
+    const result = { total: 0, time: 0 };
```

Comment:
I get that it might be slightly more efficient, but in the case the check for ensuring messages are not `undefined` is after where `result` is initialized.